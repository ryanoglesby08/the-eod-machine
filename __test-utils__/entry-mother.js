const pick = require('./pick')

const defaultEntry = {
  author: 'Some author',
  category: 'Test category',
  content: 'some content',
}

const someEntryInput = (overrides = {}) => {
  return Object.assign(
    {},
    pick(defaultEntry, ['category', 'content']),
    pick(overrides, ['category', 'content'])
  )
}

const anAuthoredEntry = (overrides = {}) => {
  return Object.assign({}, defaultEntry, overrides)
}

const someEntryInputAndAuthoredEntry = (overrides = {}) => {
  return {
    entryInput: someEntryInput(overrides),
    authoredEntry: anAuthoredEntry(overrides),
  }
}

module.exports = {
  someEntryInput,
  anAuthoredEntry,
  someEntryInputAndAuthoredEntry,
}
