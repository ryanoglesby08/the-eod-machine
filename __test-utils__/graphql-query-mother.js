const parseFields = require('graphql-parse-fields')

const pick = require('./pick')

const createMother = (query, baseObject) => {
  const fields = extractFieldsFromGraphqlQuery(query)

  return (overrides = {}) => {
    return Object.assign({}, pick(baseObject, fields), pick(overrides, fields))
  }
}

const extractFieldsFromGraphqlQuery = queryAst => {
  const fieldAst = queryAst.definitions[0].selectionSet.selections[0]
  const queryName = fieldAst.name.value

  const fieldsOfQuery = parseFields(fieldAst)

  return Object.keys(fieldsOfQuery[queryName])
}

module.exports = createMother
