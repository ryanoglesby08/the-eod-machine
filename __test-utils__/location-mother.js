const pick = require('./pick')

const baseLocation = {
  name: 'The location',
  timeZone: 'The time zone',
  eodTime: 'The EOD time',
}

const createLocationMother = relevantFields => (overrides = {}) => {
  return Object.assign(
    {},
    pick(baseLocation, relevantFields),
    pick(overrides, relevantFields)
  )
}

const aLocation = (overrides = {}) => {
  return Object.assign({}, baseLocation, overrides)
}

module.exports = { baseLocation, aLocation, createLocationMother }
