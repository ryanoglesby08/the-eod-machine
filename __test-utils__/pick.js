const pick = (object, wantedProperties) => {
  return Object.keys(object)
    .filter(property => wantedProperties.includes(property))
    .reduce((newObject, property) => {
      return Object.assign({}, newObject, { [property]: object[property] })
    }, {})
}

module.exports = pick
