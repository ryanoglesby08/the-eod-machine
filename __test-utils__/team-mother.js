const pick = require('./pick')
const { anAuthoredEntry } = require('./entry-mother')

const defaultTeam = {
  _id: 'team-0',
  name: 'The team',
  mailingList: ['team@example.com'],
  currentEod: [anAuthoredEntry()],
}

const aTeamWithItsEod = (overrides = {}) => {
  return Object.assign({}, defaultTeam, overrides)
}

const someTeamInput = (overrides = {}) => {
  return Object.assign(
    {},
    pick(defaultTeam, ['name', 'mailingList']),
    overrides
  )
}

module.exports = { aTeamWithItsEod, someTeamInput }
