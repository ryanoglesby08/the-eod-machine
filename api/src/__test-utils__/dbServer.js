import { MongoMemoryServer } from 'mongodb-memory-server'

const mongoServer = new MongoMemoryServer()

export const startDbServer = async () => ({
  server: mongoServer,
  uri: await mongoServer.getConnectionString(),
})

export const stopDbServer = async () => {
  await mongoServer.stop()
}
