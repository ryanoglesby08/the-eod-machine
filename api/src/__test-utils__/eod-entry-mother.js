const defaultEntry = {
  category: 'Test category',
  content: 'some content',
}

const anEntry = (overrides = {}) => {
  return Object.assign({}, defaultEntry, overrides)
}

module.exports = anEntry
