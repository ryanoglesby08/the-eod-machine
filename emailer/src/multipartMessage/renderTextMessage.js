import textMessage from './textMessage/textMessage'

const renderTextMessage = team => textMessage(team.currentEod)

export default renderTextMessage
