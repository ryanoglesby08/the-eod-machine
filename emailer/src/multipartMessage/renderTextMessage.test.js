import { aTeamWithItsEod } from '../../../__test-utils__/team-mother'
import { anAuthoredEntry } from '../../../__test-utils__/entry-mother'

import renderTextMessage from './renderTextMessage'

it('is an EOD message as plain text', () => {
  const team = aTeamWithItsEod({
    currentEod: [
      anAuthoredEntry({
        author: 'Some author',
        category: 'Category 1',
        content: 'some content',
      }),
      anAuthoredEntry({
        author: 'Another author',
        category: 'Category 1',
        content: 'more content',
      }),
      anAuthoredEntry({
        author: 'Some author',
        category: 'Category 2',
        content: 'even more content',
      }),
    ],
  })

  const message = renderTextMessage(team)

  expect(message).toMatchSnapshot()
})

it('is an empty EOD message when there are no entries', () => {
  const team = aTeamWithItsEod({
    currentEod: [],
  })

  const message = renderTextMessage(team)

  expect(message).toMatchSnapshot()
})
