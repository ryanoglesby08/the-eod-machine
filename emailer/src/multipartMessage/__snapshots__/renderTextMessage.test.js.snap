// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is an EOD message as plain text 1`] = `
"
EOD Updates

---------------------------------------------------------------------
Category 1

* some content (Some author)
* more content (Another author)

---------------------------------------------------------------------
Category 2

* even more content (Some author)


- Delivered by The EOD Machine
"
`;

exports[`is an empty EOD message when there are no entries 1`] = `
"
EOD Updates

No updates this time.


- Delivered by The EOD Machine
"
`;
